### ISSUE TITLE
[improvement] Display geolocated photos on a map

### ISSUE THREAD

**Issue by asitemade4u** [xemle/home-gallery#33]
HG is really good at reading photo metadata.
It would be great to use the geolocation metadata to display thumbnails of photos on a map, like digiKam and PhotoPrism do.

**Comment by xemle** [comment]
Yes, indeed. This feature is missing and I could use it, too.

The reason why this is missing: it was not yet important for me on daily basis. It would be gorgeous for presenting my demo page with all the places I have been and would have a massive wow effect. The effort for the wow effect is not small.

This feature is on my list after the automatism actions, described in #35.

**Comment by xemle** [comment]
I am testing the new map feature for geo located photos. Please have a look at the [demo map](https://demo.home-gallery.org/map) and leave comments here

**Comment by darrepac** [comment]
Hi

Sounds good!
1/ How do you see the pictures of a cluster? Sometimes you can zoom it to differentiate, sometimes not like here:



2/ when you see a photo full screen you cannot go back to map view but only to the classical view (or I didn’t find it)

**Comment by xemle** [comment]
Thank you @darrepac for taking the time.

1) I've updated the [demo map](https://demo.home-gallery.org/map) to see some preview of the cluster. When you click to a cluster, up to 16 squared small preview images are shown with the date range. You can click on the cluster and see only images of the cluster

2) You can enter the map from an image by showing the details via the info icon. There you can click on the geo location and the map opens.

If you are on the map and open an image and you can see the full screen image. If you click on the stream image you should see the stream with the selected image.

**Comment by MarioSob** [comment]
As this feature looks already developed (looking at demo map) - are there any chances releasing it soon?

**Comment by xemle** [comment]
Hi @MarioSob 

Thank you for the request

TL;DR I'm on it.

Since the used [leaflet react](https://react-leaflet.js.org) library for the map requires react 18 I am also upgrading react. This leads to further migration tasks of other libs. It is a bit of yak shaving and takes more time.

I could provide the map development branch to github and as binary if that helps?

**Comment by MarioSob** [comment]
Hi @xemle It's not that urgent. I was only afraid the interest in developing this is lost and that would be a shame at such advanced stage like in demo gallery. I use docker image anyway so full release with image built is what I'm interested the most. I'm happy to hear it's on its way. Also I slowly started probing feature sets of other galleries just to have this single feature. I wanted to know if it makes sense looking for alternatives if I can simply wait for next release.
Thank you.

**Comment by xemle** [comment]
Hi @MarioSob 

Thank you for your quick and hones reply

> I use docker image anyway so full release with image built is what I'm interested the most.

I published the WIP [feature/webapp-map](https://github.com/xemle/home-gallery/tree/feature/webapp-map) branch. Than you can build your own docker image based on the branch (see [docs](https://docs.home-gallery.org/install/index.html#custom-image) for hints) and test the map feature to check if it fits your needs. ...and you can provide some valuable feedback.

If you have further questions, please drop a note

**Comment by MarioSob** [comment]
Thank you, @xemle 
I actually failed on using the application form the branch. I've cloned out repo from git, checked out the branch you prepared, performed build which finished with "Successfully tagged home-gallery:latest" but after all docker-compose downloaded and used official image ignoring the whole process... I don't know how to force it to use the image I've just built.
BTW: [docs](https://docs.home-gallery.org/install/index.html#custom-image) should instruct to use "git clone" instead of "git checkout".

**Comment by xemle** [comment]
> Thank you, @xemle I actually failed on using the application form the branch. I've cloned out repo from git, checked out the branch you prepared, performed build which finished with "Successfully tagged home-gallery:latest"

Congratulations! Good job

> but after all docker-compose downloaded and used official image ignoring the whole process... I don't know how to force it to use the image I've just built.

Maybe you need to use your local docker image by `home-gallery` instead of the public official `xemle/home-gallery`. So please update your `docker-compose.yml`.

> BTW: [docs](https://docs.home-gallery.org/install/index.html#custom-image) should instruct to use "git clone" instead of "git checkout".

Thank you. I will fix this typo next time.

> BTW2: Suggestion on demo gallery: I think there is no point in showing "Show map of entry" button on pictures that have no GPS tags in EXIF (so clicking it does nothing).

True. Thank you for that feedback. I will also honor this next iteration.

**Comment by xemle** [comment]
Hi @MarioSob 

> BTW2: Suggestion on demo gallery: I think there is no point in showing "Show map of entry" button on pictures that have no GPS tags in EXIF (so clicking it does nothing).

I can not follow you with your description. Could you send a screenshot if possible?

**Comment by MarioSob** [comment]
Hi @xemle 
Thank you for publishing new version, that makes things easier. I will provide you more details soon as it turned out my destination environment is malfunctioning and I need to put it back together first. :-/
